app/models/document.rb

```diff
@@ -0,0 +1,7 @@
+class Document < ApplicationRecord
+  belongs_to :user
+  belongs_to :assistant, optional: true
+  belongs_to :message, optional: true
+
+  enum purpose: %w[ fine-tune fine-tune-results assistants assistants_output ].index_by(&:to_sym)
```

Comment:
I find it easier to deal with enums when the underlying field is a string rather than an integer. It makes declaring them a little more gnarly with the index_by() thingy, but besides that, this is just a normal activerecord enum field.